feat: 로그인/리다이렉트, NotFound, 랭킹 필터 상태 유지 등 기능 구현

hyemomo opened the pull request:
### 로그인 UI 구현
- `/login` 경로 접속 시 로그인 페이지로 진입
-  로그인 페이지로 리다이렉팅, 이전 페이지로 리다이렉팅 하기 구현
### Not Found
 - 존재하지 않는 경로이면 모두 Not Found 페이지로 가도록 연결
### 필터 유지 기능
- state로 관리했던 필터를 쿼리 파라미터로 관리함
- 새로 고침 시에도 유지
### 기타
- 이전 pr 리뷰 피드백을 반영해 type을 더 명확하게 함

hyemomo committed:
feat: 절대경로 alias 설정

hyemomo committed:
style: Prettier, Emotion, Pretendard 기본 스타일 환경 세팅

hyemomo committed:
docs: 초기 세팅 단계 정리한 README 추가

hyemomo committed:
style: emotion에 Color, Typography 토큰  설정

hyemomo committed:
style: 스타일링 초기화

hyemomo committed:
feat: navbar UI 구현

hyemomo committed:
chore: 추가 설정 및 초기화

hyemomo committed:
feat: 친구 선택 박스 UI 구현

hyemomo committed:
feat: 선물 테마 카테고리 UI 구현

hyemomo committed:
feat: mobile first design을 위한 breakpoint 및 container 컴포넌트 추가

hyemomo committed:
chore: 사용하지 않는 파일 삭제

hyemomo committed:
배너 UI 구현, Padding 컴포넌트 크기 추가

hyemomo committed:
feat: 선물랭킹 UI 구현

hyemomo committed:
feat: navbar 아이콘 수정

hyemomo committed:
feat: styled-component 스타일링 수정

hyemomo committed:
style: container의 임시 배경 수정, button 스타일링 초기화, App.tsx 수정

hyemomo committed:
Merge branch 'hyemomo' into step2

hyemomo committed:
feat: svg 이미지 따로 관리

hyemomo committed:
style: svg 이미지 삽입, 개행문자와 포맷팅 수정, 코드 실수 수정

hyemomo committed:
feat: ranking 컴포넌트 분리, 더보기 기능 구현

hyemomo committed:
줄바꿈 정리

hyemomo committed:
feat: 필터 기능 구현

hyemomo committed:
feat: numeric separator로 숫자 변경

hyemomo committed:
feat: 로그인 페이지 UI/라우팅 구현

hyemomo committed:
feat: Not Found 페이지 구현

hyemomo committed:
feat: query parameters을 이용한 필터 상태 유지

hyemomo committed:
feat: PR 리뷰 반영 - 필터 옵션 타입 명확하게 지정)

hyemomo committed:
Merge branch 'hyemomo' into step3

yujo11 commented on src/components/navbar/Navbar.tsx:
```diff
@@ -1,26 +1,37 @@
-        padding: 0 ${({theme})=>theme.spacing.spacing3};
-        height: 2.75rem;
-        background-color: ${({theme})=>theme.colors.gray};`
-  const PreviousButton = styled.button`
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    padding: 0 ${({ theme }) => theme.spacing.spacing3};
+    height: 2.75rem;
+    background-color: ${({ theme }) => theme.colors.gray};
+  `;
+  const Button = styled.button`
     width: ${({ theme }) => theme.spacing.spacing7};
     height: ${({ theme }) => theme.spacing.spacing7};
   `;
```
styled 코드들이 `Navbar` 컴포넌트 내부에 위치하고 있는 이유가 있나요?

yujo11 commented on src/components/navbar/Navbar.tsx:
```diff
@@ -1,26 +1,37 @@
       <div>
-        <img src={myIcon} alt="" />
+        <Button onClick={()=>{handleLink("/login")}}>
+          <img src={myIcon} alt="" />
```
alt 태그도 비워두지 말고 잘 챙겨주시면 좋을거 같네요 ㅎㅎ

yujo11 commented on src/components/navbar/Navbar.tsx:
```diff
@@ -1,26 +1,37 @@
       <div>선물하기</div>
       <div>
-        <img src={myIcon} alt="" />
+        <Button onClick={()=>{handleLink("/login")}}>
```
login 같은 path들은 상수화해서 사용해보시면 좋을거 같아요

yujo11 commented on src/components/navbar/Navbar.tsx:
```diff
@@ -1,26 +1,37 @@
     <NavbarWrapper>
       <div>
-        <PreviousButton><img src={arrowLeftIcon} alt="" /></PreviousButton>
+        <Button onClick={()=>{navigate(-1)}}>
```
유저가 다른 사이트에 있다가 선물하기 사이트 url로 접근한 경우에 `navigate(-1)`은 유저를 어디로 보낼까요?

yujo11 commented on src/pages/Login.tsx:
```diff
@@ -0,0 +1,68 @@
+  width: 100%;
+  height: 2.75rem;
+`;
+const Login = () => {
+    const navigate=useNavigate();
+  return (
```
코드 포맷터가 적용이 잘 안 되고 있는거 같네요 👀

yujo11 commented on src/routes/Router.tsx:
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { BrowserRouter, Routes, Route } from "react-router-dom";
+
+const Router = () => {
```
최신버전에서 권장되는 `createBrowserRouter`를 이용해보셔도 좋을거 같아요

yujo11 commented on src/components/ranking/types.ts:
```diff
@@ -1,17 +1,38 @@
+  "여자가",
+  "청소년이",
+] as const;
+export type PersonFilterLabels = (typeof PERSON_FILTER_LABELS)[number];
```
👍 👍 👍

yujo11 reviewed (changes requested):
안녕하세요 혜민님 ~~

소소한 코멘트들 추가로 남겼는데 확인해주시면 좋을거 같습니다!

고생 많으셨습니다 ~~ 👍 👍 👍

hyemomo committed:
refactor: 피드백 반영
- createBrowserRouter + RouterProvider 적용
- img alt 속성 추가
- 이전 페이지 이동시 외부사이트에서의 이동까지 고려
- path 상수화

hyemomo commented:
피드백 반영해서 코드 수정했습니다. 리뷰 요청 확인을 하면서 실수가 있었어서 미션1 리뷰재요청 드립니다. 죄송합니다.

yujo11 commented on src/routes/Router.tsx:
```diff
@@ -0,0 +1,27 @@
+import Login from "@/pages/Login";
+import Main from "@/pages/Main";
+import NotFound from "@/pages/NotFound";
+import { createBrowserRouter } from "react-router-dom";
+//TODO: createBrowserRouter 사용해보기
+export const ROUTE_PATH = {
+  HOME: "/",
+  LOGIN: "/login",
+  NOT_FOUND: "*",
+};
+const router = createBrowserRouter([
```
👍 👍 👍

yujo11 replied:
소소하게 이 사이에 개행이 하나씩 들어가면 코드 보기가 한결 편해질거 같습니다!

yujo11 commented on src/routes/Router.tsx:
```diff
@@ -0,0 +1,27 @@
+import Main from "@/pages/Main";
+import NotFound from "@/pages/NotFound";
+import { createBrowserRouter } from "react-router-dom";
+//TODO: createBrowserRouter 사용해보기
```
해소 된 TODO는 제거해주셔도 좋을거 같네요 ㅎㅎ

yujo11 reviewed (approved):
안녕하세요 헤민님 ~ 이번 미션은 여기서 마무리해도 좋을거 같네요!

미션 진행하느라 고생 많으셨습니다 👍 👍 👍
